client/views/view_research.cpp

```diff
@@ -500,6 +512,12 @@ void science_report::update_report()
     researching_combo->setDisabled(false);
     goal_combo->setDisabled(false);
   }
+
+  // If tech leak happens we enable a button to force/push a refresh.
+  if (done >= total) {
+    refresh_but->setVisible(true);
```

Comment:
Just out of curiosity, are you changing the visibility of the button here instead of disabling it for a reason?

I'm asking because I personally find it a bit confusing when controls are sometimes visible and sometimes not.

Reply:
I also think hiding a control is bad practice (the exception we make is when the thing it controls is disabled in the rules, as is done for controls related to nationality). I would prefer to keep it visible at all times. In fact, I don't think even disabling this button is needed: refreshing is harmless when it doesn't do anything.